packages/sdk-v2/src/sdk/goodid/hooks/useCertificates.ts

```diff
@@ -1,28 +1,71 @@
-import { useCallback } from "react";
-import { isEmpty } from "lodash";
-import { CredentialTypes } from "../types";
+import { useCallback, useContext, useEffect, useState } from "react";
+import { first } from "lodash";
 
 import type { Certificate } from "../types";
 
-// This expects a single issuer (gd) for the mvp purpose
-export const useCertificates = (db: any) => {
-  const getCertificates: (credentialTypes?: string[]) => Promise<Certificate[] | undefined> = useCallback(
-    async credentialTypes => {
-      const keys = credentialTypes ?? Object.keys(CredentialTypes);
-      return await db.certificates.where("type").anyOf(keys).toArray();
+import { GoodIdContext } from "../../../contexts";
+
+const cleanupCertificate = (value: any) => {
+  if (!value) return [];
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { primary_idx, type_subject_idx, ...certificate } = value;
+  return certificate;
+};
+
+const queryCertificates = async (db: any, account: string | undefined, types: string[] | null = null) => {
+  if (!db || !account) {
+    throw new Error("Certificates database not initialized");
+  }
+
+  const accountFilter = `did:ethr:${account}`;
+  const queryField = types ? "type_subject_idx" : "credentialSubject.id";
+  const queryValue = types ? types.map(type => `${accountFilter}_${type}`) : accountFilter;
+  const queryOperator = types ? "anyOfIgnoreCase" : "equalsIgnoreCase";
+  return db.certificates.where(queryField)[queryOperator](queryValue).reverse().sortBy("issuanceDate");
+};
+
+// set types to `false` explicitly to avoid loading on mount
+// passign null or empty value will just load all certs without type filtering
+export const useCertificates = (account: string | undefined, types: string[] | null | false = null) => {
+  const [certificates, setCertificates] = useState<Certificate[]>();
+  const { db } = useContext(GoodIdContext);
+
+  const getCertificate = useCallback(
+    async (type: string[]) =>
+      queryCertificates(db, account, type)
+        .then(first) // from lodash
+        .then(cleanupCertificate),
+    [account, db]
+  );
+
+  const storeCertificate = useCallback(
+    async certificate => {
+      const { type, credentialSubject } = certificate;
+      const typeSubjectIdx = `${credentialSubject.id}_${type[1]}`;
+
+      // unfortunately we need an additional manual-filled field for the index
+      // because compound indexes cannot be combined with multi-entry ones
+      // that's IndexedDB limitation (
+      return db.certificates.add({ ...certificate, type_subject_idx: typeSubjectIdx });
```

Comment:
without having a unique identifiable primary key this will just keep continue adding an maybe already existing credential.
To me it would make more sense to have the type_subject_idx as primary and use put (add when not existing/updating when existing).

Would not need any additional filter on issuance date then

Reply:
that's ok, if you issue new certificate it will be used instead of the old one

Reply:
you could add check does **the same** certificate added by comparing some fields (I suppose  jwt would be enough or issuer + date to faster indexing/querying).

but. I really don't think we need this for MVP